docs/user_guide/getting_started/minimal_script_noapp.rst

```diff
@@ -0,0 +1,273 @@
+.. currentmodule:: globus_sdk
+
+.. _minimal_script_noapp_tutorial:
+
+Creating A Script without GlobusApp
+===================================
+
+:class:`GlobusApp` provides a number of useful abstractions in the SDK.
+It handles login flows and storage of tokens, coupled with later retrieval of
+those tokens for use. It can keep track of which clients have been created and
+registered with an app, and therefore make intelligent decisions about how and
+when to prompt users to login.
+
+New users should read :ref:`the guide for writing a minimal script
+<minimal_script_tutorial>` before reading this doc.
+
+:class:`GlobusApp` is built from several simpler components which can be used to
+implement similar behaviors.
+This doc covers how to write a simple script, but without using
+:class:`GlobusApp`.
+
+For readers who prefer to start with complete examples, jump ahead to these
+sections before reviewing the doc:
+
+-   :ref:`Login and List Groups (simple) <list_groups_noapp>`
+-   :ref:`Login and List Groups (with storage) <list_groups_noapp_with_storage>`
+
+Cases for not Using GlobusApp
+-----------------------------
+
+There are at least three main reasons to be interested in defining applications
+without :class:`GlobusApp`:
+
+1.  You have a use case which doesn't fit the behaviors of an app.
+    e.g., Implementations of *APIs* or *services*.
+
+2.  Any legacy codebase, predating :class:`GlobusApp`, will use the underlying
+    constructs to implement login behaviors.
+
+3.  Customizing and extending :class:`GlobusApp` to suit your use case may
+    require understanding the underlying components.
+
+.. note::
+
+    Prior to the introduction of :class:`GlobusApp`, these tools were the only
+    way that an application could be written with the ``globus_sdk``. If you
+    are maintaining an existing application, you may need to be strategic about
+    when and how to upgrade such usages.
```

Comment:
Tangential thought: this note makes me wonder if we should have a migration guide for adopting GlobusApp that we can reference for case #2.

Reply:
I think we need to write such a doc, but it doesn't exist yet. Part of why I'm eager to land this, to clear the way.